src/sentry/static/sentry/app/components/resultGrid.tsx

```diff
@@ -129,103 +142,113 @@ class SortBy extends React.Component {
   }
 }
 
-const ResultGrid = createReactClass({
-  displayName: 'ResultGrid',
-
-  propTypes: {
-    api: PropTypes.object,
-    columns: PropTypes.array,
-    columnsForRow: PropTypes.func,
-    defaultSort: PropTypes.string,
-    defaultParams: PropTypes.object,
-    endpoint: PropTypes.string,
-    filters: PropTypes.object,
-    hasPagination: PropTypes.bool,
-    hasSearch: PropTypes.bool,
-    keyForRow: PropTypes.func,
-    location: PropTypes.object,
-    method: PropTypes.string,
-    options: PropTypes.array,
-    path: PropTypes.string,
-    sortOptions: PropTypes.array,
-  },
-
-  getDefaultProps() {
-    return {
-      path: '',
-      endpoint: '',
-      method: 'GET',
-      columns: [],
-      sortOptions: [],
-      filters: {},
-      defaultSort: '',
-      keyForRow: function (row) {
-        return row.id;
-      },
-      columnsForRow: function () {
-        return [];
-      },
-      defaultParams: {
-        per_page: 50,
-      },
-      hasPagination: true,
-      hasSearch: false,
-    };
-  },
-
-  getInitialState() {
-    const queryParams = (this.props.location || {}).query || {};
+type FilterConfig = {
+  name: string;
+  options: Option[];
+};
+
+type Props = {
+  api: Client;
+  location: Location;
+} & DefaultProps;
+
+type DefaultProps = {
+  columns: React.ReactNode[];
```

Comment:
wouldn't it be better if we reaplce `React.ReactNode[]`; by `React.ReactElement[]`; ?

Reply:
Yeah I think this is OK though